src/dotnet/commands/dotnet-add-p2p/Program.cs

```diff
@@ -0,0 +1,235 @@
+                {
+                    throw new GracefulException(Strings.RequiredArgumentNotPassed, "<Project>");
+                }
+
+                ProjectRootElement project;
+                string projectDir;
+                if (File.Exists(projectArgument.Value))
+                {
+                    project = GetProjectFromFileOrThrow(projectArgument.Value);
+                    projectDir = new FileInfo(projectArgument.Value).DirectoryName;
+                }
+                else
+                {
+                    project = GetProjectFromDirectoryOrThrow(projectArgument.Value);
+                    projectDir = projectArgument.Value;
+                }
+
+                projectDir = PathUtility.EnsureTrailingSlash(projectDir);
+
+                if (app.RemainingArguments.Count == 0)
+                {
+                    throw new GracefulException(Strings.SpecifyAtLeastOneReference);
+                }
+
+                List<string> references = app.RemainingArguments;
+                if (!forceOption.HasValue())
+                {
+                    EnsureAllReferencesExist(references);
+                    ConvertPathsToRelative(projectDir, ref references);
+                }
+                
+                int numberOfAddedReferences = AddProjectToProjectReference(
+                    project,
+                    frameworkOption.Value(),
+                    references);
+
+                if (numberOfAddedReferences != 0)
+                {
+                    project.Save();
+                }
+
+                return 0;
+            });
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (GracefulException e)
+            {
+                Reporter.Error.WriteLine(e.Message.Red());
+                app.ShowHelp();
+                return 1;
+            }
+        }
+
+        internal static void EnsureAllReferencesExist(List<string> references)
+        {
+            var notExisting = new List<string>();
+            foreach (var r in references)
+            {
+                if (!File.Exists(r))
+                {
+                    notExisting.Add(r);
+                }
+            }
+
+            if (notExisting.Count > 0)
+            {
+                throw new GracefulException(
+                    string.Join(
+                        Environment.NewLine,
+                        notExisting.Select((r) => string.Format(Strings.ReferenceDoesNotExist, r))));
+            }
+        }
+
+        internal static void ConvertPathsToRelative(string root, ref List<string> references)
+        {
+            root = PathUtility.EnsureTrailingSlash(Path.GetFullPath(root));
+            references = references.Select((r) => PathUtility.GetRelativePath(root, Path.GetFullPath(r))).ToList();
+        }
+
+        // There is ProjectRootElement.TryOpen but it does not work as expected
+        // I.e. it returns null for some valid projects
```

Comment:
ProjectRootElement.TryOpen is document that it is trying to open from the cache as oposed to trying to open the file